### ISSUE TITLE
no-restricted-paths doesn't work 

### ISSUE THREAD

**Issue by jonasbjoralt** [import-js/eslint-plugin-import#2564]
Hi, I have the following structure:

```
project
├── .eslintrc.js
├── generated
        └── foo.ts
└── src
        └── components
                └── MyComponent.tsx
``` 

And the following rule in `.eslintrs.js`:
```
  rules: {
    "import/no-restricted-paths": [
      "error",
      {
        zones: [
          {
            target: "./src",
            from: "./generated",
          },
        ],
      },
    ]
}
```

However, this import from `src/` which should give an error doesn't: 
```
// src/components/MyComponent.tsx
import { bar } from "../../../generated/foo";
```

Any ideas?

**Comment by ljharb** [comment]
Can you share a repro repo?

(also is it eslintrs or eslintrc)

**Comment by greim** [comment]
Attaching a repro.

It seems related to typescript eslint. If I remove it and convert the repro to pure .js, it works.

[no-restricted-paths-testcase.zip](https://github.com/user-attachments/files/19489010/no-restricted-paths-testcase.zip)

**Comment by JounQin** [comment]
Please provide a GitHub repository reproduction instead.

**Comment by greim** [comment]
https://github.com/greim/no-restricted-paths-repro

**Comment by greim** [comment]
On further investigation, my issue may be resolved via [these instructions](https://github.com/import-js/eslint-plugin-import/tree/main?tab=readme-ov-file#typescript) which I missed originally.

@jonasbjoralt - Not sure if you tried that but might be worth checking. Looks like your example used TypeScript also.

**Comment by ljharb** [comment]
Closing as resolved, but happy to reopen if that doesn't solve it.
